### PR TITLE
Modifying Insertion Sort [Python]

### LINKED ISSUE #259: Insertion Sort [Python]
Bad implementation (i.e not working). The `while` loop [here](https://github.com/iiitv/algos/blob/master/insertion_sort/insertion_sort.py#L9) should be indented one tab right (should be inside the `for` loop).

Description Insertion Sort: https://en.wikipedia.org/wiki/Insertion_sort

### PR DESCRIPTION
**Pull request by TarunISCO**
Fixes #259

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
@TarunISCO Remove .bin file from this commit.
@aviaryan @singhpratyush I think `*.bin` should also be added to .gitignore

**Comment by TarunISCO** [comment]
from where? @aashutoshrathi

**Comment by aashutoshrathi** [comment]
@TarunISCO, these 2 files
screenshot 20

File: breadth_first_search/BreadhFirstSearch.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Program to print Breadth First Traversal of a Graph
+// Time Complexity : O( no_of_vertices + no_of_edges)
+// Space Complexity : O( no_of_vertices + no_of_edges)
```
Comment (aviaryan) [inline breadth_first_search/BreadhFirstSearch.cpp]: Remove BFS files from this PR.

File: README.md [hunk 2]
```diff
@@ -15,28 +15,45 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 
 | Algorithm | C | CPP | Java | Python | Golang | JavaScript |
 |:--------------|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|:-----------------:|
+<<<<<<< HEAD
```
Comment (Monal5031) [inline README.md]: Please remove these unnecessary lines 18-20 and 22.

File: README.md [hunk 3]
```diff
@@ -15,28 +15,45 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Coin Change Problem](http://www.algorithmist.com/index.php/Coin_Change) | [\[X\]](coin_change_problem/coin_change_problem.c) | | | | | |
 | [Counting Sort](http://www.geeksforgeeks.org/counting-sort/)| [\[X\]](counting_sort/counting_sort.c) | | [\[X\]](counting_sort/CountingSort.java) | | | |
 | [Depth First Traversal](http://www.geeksforgeeks.org/depth-first-traversal-for-a-graph/) | | | [\[X\]](depth_first_traversal/DepthFirstTraversal.java) | | | |
 | [Dijkstra Algorithm](https://en.wikipedia.org/wiki/Dijkstra's_algorithm/) | [\[X\]](dijkstra/dijkstra.c) | | [\[X\]](dijkstra/Dijkstra.java) | | | |
 | [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm) | [\[X\]](euclidean_gcd/euclidean_gcd.c) | | [\[X\]](euclidean_gcd/EuclideanGCD.java) | [\[X\]](euclidean_gcd/euclidean_gcd.py) | | |
 | [Exponentation by Squaring](https://en.wikipedia.org/wiki/Exponentiation_by_squaring) | [\[X\]](exponentation_by_squaring/exponentation_by_squaring.c) |||| | |
+<<<<<<< HEAD
```
Comment (Monal5031) [inline README.md]: Also these 30-33 and 36

File: README.md [hunk 4]
```diff
@@ -15,28 +15,45 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Largest Sum Contiguous Subarray](http://www.geeksforgeeks.org/largest-sum-contiguous-subarray/) | [\[X\]](largest_sum_contiguous_subarray/largestSumContiguousSubarray.c) | | [\[X\]](largest_sum_contiguous_subarray/LargestSumContiguousSubarray.java) | | [\[X\]](largest_sum_contiguous_subarray/largestSumContiguousSubarray.go) | |
 | [Linear Search](https://en.wikipedia.org/wiki/Linear_search) | [\[X\]](linear_search/linear_search.c) | | [\[X\]](linear_search/LinearSearch.java) | [\[X\]](linear_search/linear_search.py) | [\[X\]](linear_search/linear-search.go) | |
 | [Longest Common Subsequence](http://www.geeksforgeeks.org/dynamic-programming-set-4-longest-common-subsequence) | [\[X\]](longest_common_subsequence/longestCommonSubsequence.c) | | [\[X\]](longest_common_subsequence/LongestCommonSubsequence.java) | | [\[X\]](longest_common_subsequence/longestCommonSubsequence.go) | |
 | [Merge Sort](https://www.khanacademy.org/computing/computer-science/algorithms/merge-sort/a/overview-of-merge-sort) | [\[X\]](merge_sort/merge_sort.c) | | [\[X\]](merge_sort/MergeSort.java) | [\[X\]](merge_sort/merge_sort.py) | | |
+<<<<<<< HEAD
```
Comment (Monal5031) [inline README.md]: Also these 42-44 and 46

File: README.md [hunk 5]
```diff
@@ -15,28 +15,45 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
+=======
 | [Modular Exponential](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/) | [\[X\]](modular_exponential/modular_exponential.c) | | [\[X\]](modular_exponential/ModularExponential.java) | [\[X\]](modular_exponential/modular_exponential.py) | [\[X\]](modular_exponential/modular_exponential.go) | |
+>>>>>>> 7fdb1bfefa2ec9880c65cd470224fde4a3a4e811
 | [N-Queen Problem](https://en.wikipedia.org/wiki/Eight_queens_puzzle) | | [\[X\]](n_queen_problem/NQueenProblem.cpp) | [\[X\]](n_queen_problem/NQueenProblem.java) | | | |
 | [Prime Factor](https://en.wikipedia.org/wiki/Prime_factor) | [\[X\]](prime_factor/prime_factor.c) | | | | | |
 | [Prims](https://en.wikipedia.org/wiki/Prim%27s_algorithm) | [\[X\]](prims/prims.c) | | [\[X\]](prims/Prims.java) | | | |
 | [Quick Select](https://en.wikipedia.org/wiki/Quickselect) | [\[X\]](quick_select/quick_select.c) | | | | | |
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | [\[X\]](quicksort/QuickSort.java) | [\[X\]](quicksort/quick_sort.py) | | |
+<<<<<<< HEAD
```
Comment (Monal5031) [inline README.md]: also 52-54 and 56

File: README.md [hunk 6]
```diff
@@ -47,6 +64,10 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 |:--------------|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|:-----------------:|
 | [AVL Tree](http://www.geeksforgeeks.org/avl-tree-set-1-insertion)| | |[\[X\]](avl_tree/AvlTree.java) | | | |
 | [Binary Search Tree](https://en.wikipedia.org/wiki/Binary_search_tree) | | | [\[X\]](binary_search_tree/BinarySearchTree.java) | | | |
+<<<<<<< HEAD
```
Comment (Monal5031) [inline README.md]: also 68-70

File: README.md [hunk 7]
```diff
@@ -61,6 +82,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | Python |<pre>python [filename.py]</pre>|
 | Golang |<pre>go run [filename.go]</pre>|
 | JavaScript |<pre>node [filename.js]</pre>|
+>>>>>>> 7fdb1bfefa2ec9880c65cd470224fde4a3a4e811
```
Comment (Monal5031) [inline README.md]: also this line

File: insertion_sort/insertion_sort.py [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
-        position -= 1
-        alist[position] = current_value
+	"""
+	Performs a insertion sort
+	:type alist: object
+	"""
+	for index in range(1, len(alist)):
+		current_value = alist[index]
+		position = index
+		while position > 0 and alist[position-1] > current_value:
```
Comment (Monal5031) [inline insertion_sort/insertion_sort.py]: Please add space around binary operator.

File: insertion_sort/insertion_sort.py [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
+			alist[position] = alist[position-1]
```
Comment (Monal5031) [inline insertion_sort/insertion_sort.py]: Please add space around binary operator.

**Comment by TarunISCO** [comment]
Why did Travis CI build failed? @singhpratyush

**Comment by Monal5031** [comment]
@TarunISCO travis is failing because you have used tab indentation instead of 4 spaces. You can see the log here https://travis-ci.org/iiitv/algos/builds/242709259?utm_source=github_status&utm_medium=notification it is mentioned there.

File: README.md [hunk 10]
```diff
@@ -63,6 +63,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | JavaScript |<pre>node [filename.js]</pre>|
 
 
+
```
Comment (Monal5031) [inline README.md]: Please remove this line, this is not part of your PR.

**Pull request merged**